src/TLuaInterpreter.cpp

```diff
@@ -5775,13 +5777,104 @@ int TLuaInterpreter::sendTelnetChannel102(lua_State* L)
     return 1;
 }
 
+// Internal helper function for two following functions:
+// returns 0 on success and sets pItem to point to the TAction with the ID or
+//   the first one found with the name as the index item.
+// returns a non-zero number on failure (of the number of items on the stack
+//   for the caller to return to ITS caller) and sets those items,
+//   it also resets the pItem to be a nullptr
+// does NOT return (normally) if the index item on the stack is not a number or
+//   a string
+int TLuaInterpreter::getTActionFromIdOrName(lua_State* L, TAction** pItem, const int index, const char* func)
```

Comment:
I know this isn't atypical for C++, but I prefer using function arguments as input instead of output. This is also mentioned in Clean Code. Have you considered returning multiple values via https://www.geeksforgeeks.org/returning-multiple-values-from-a-function-using-tuple-and-pair-in-c/?

Reply:
Agree - and we have many examples of this, check for example:

```cpp
    std::pair<bool, QString> changeModuleSync(const QString& enableModuleName, const QLatin1String &value);
    std::pair<bool, QString> getModuleSync(const QString& moduleName);
```